controller/src/test/java/io/pravega/controller/server/eventProcessor/ScaleRequestHandlerTest.java

```diff
@@ -293,7 +293,7 @@ public void testScaleWithTransactionRequest() throws InterruptedException {
                 Lists.newArrayList(new AbstractMap.SimpleEntry<>(0.0, 1.0)), false, System.currentTimeMillis(), System.currentTimeMillis())).join();
 
         // 3. verify that scale is complete
-        State state = streamStore.getState(scope, stream, false, null, executor).join();
+        State state = streamStore.getState(scope, stream, true, null, executor).join();
```

Comment:
I don't have a good sense of how often we end up ignoring the cache in the code. I wonder if we do it often enough that it doesn't make sense to have a cache at all.

Reply:
We use the cache for all immutable records and mutable records for getters. 
For example if multiple clients request for head or tail of the stream, they are all served from the cache. 
So truncationRecord for head and currentEpochRecord for tail are cached and these getters use ignoreCached = false. 

For all workflows, we always retrieve the value from the store and do not rely on cached value.